tests/system/test_irqbalance.py

```diff
@@ -47,11 +52,13 @@ def test_start_four_vms(self, four_vms):
             # List the CPU(s) that handled IRQs for the VM's vifs
             for vif in vm.vifs():
                 device_id = vif.device_id()
-                output = vm.host.ssh([rf'grep /proc/interrupts -e "xen-dyn\\s\+-event\\s\+{device_id}-"'])
+                # depending on kernel patches, we're looking either for xen-dyn or xen-dyn-lateeoi
+                output = vm.host.ssh([rf'grep /proc/interrupts -e "xen-dyn\(-lateeoi\)\?\\s\+-event\\s\+{device_id}-"'])
                 assert len(output) > 0
                 for line in output.splitlines():
                     fields = line.split()
-                    irqs_per_cpu = fields[1:fields.index('xen-dyn')]
+                    xen_dyn_index = fields.index('xen-dyn') if 'xen-dyn' in fields else fields.index('xen-dyn-lateeoi')
```

Comment:
```python
try:
    xen_dyn_index = fields.index('xen-dyn')
except ValueError:
    xen_dyn_index = fields.index('xen-dyn-lateeoi')
```
Good habits: IMHO, it's better to check only once time for 'xen-dyn'. :)

Reply:
Or better: ```python
next((i for i, item in enumerate(fields) if item == 'xen-dyn'), fields.index('xen-dyn-lateeoi'))
```

Reply:
As we discussed, I find the solution with `next` less readable for a negligible gain. I used your the try except instead.